src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -132,7 +132,32 @@ internal virtual string GetAssemblyFileVersion()
             {
                 var name = $"{Type.GetTypeInfo().Assembly.GetName().Name}.dll";
                 var filePath = Path.Combine(_clientDirectory, name);
-                return FileVersionInfo.GetVersionInfo(filePath).FileVersion;
+                var fileVersionInfo = FileVersionInfo.GetVersionInfo(filePath);
+
+                return $"{fileVersionInfo.FileVersion} ({ExtractShortSHA(fileVersionInfo.ProductVersion)})";
```

Comment:
It would look like: 
`42.42.42.42424 (2.1.0-dev-56735-00. Commit Hash: 12345678b6d4d10cf04e128efc6ed56f9e424f36)`

Two problems: it is not pretty and also the `2.1....` part keeps changing so hard to test.